drivers/kw2xrf/kw2xrf.c

```diff
@@ -1100,45 +1096,57 @@ int _assemble_tx_buf(kw2xrf_t *dev, gnrc_pktsnip_t *pkt)
     /* fill in destination address */
     if (hdr->flags &
         (GNRC_NETIF_HDR_FLAGS_BROADCAST | GNRC_NETIF_HDR_FLAGS_MULTICAST)) {
-        dev->buf[2] = 0x88;
+        dev->buf[2] = IEEE802154_FCF_DST_ADDR_SHORT;
         dev->buf[index++] = 0xff;
         dev->buf[index++] = 0xff;
-        /* set source address */
-        dev->buf[index++] = (uint8_t)(dev->addr_short[0]);
-        dev->buf[index++] = (uint8_t)(dev->addr_short[1]);
     }
     else if (hdr->dst_l2addr_len == 2) {
         /* set to short addressing mode */
-        dev->buf[2] = 0x88;
+        dev->buf[2] = IEEE802154_FCF_DST_ADDR_SHORT;
         /* set destination address, byte order is inverted */
-        dev->buf[index++] = (gnrc_netif_hdr_get_dst_addr(hdr))[1];
-        dev->buf[index++] = (gnrc_netif_hdr_get_dst_addr(hdr))[0];
-        /* set source pan_id */
-        //dev->buf[index++] = (uint8_t)((dev->radio_pan) >> 8);
-        //dev->buf[index++] = (uint8_t)((dev->radio_pan) & 0xff);
-        /* set source address */
-        dev->buf[index++] = (uint8_t)(dev->addr_short[0]);
-        dev->buf[index++] = (uint8_t)(dev->addr_short[1]);
+        uint8_t *dst_addr = gnrc_netif_hdr_get_dst_addr(hdr);
+        dev->buf[index++] = dst_addr[1];
+        dev->buf[index++] = dst_addr[0];
     }
     else if (hdr->dst_l2addr_len == 8) {
         /* default to use long address mode for src and dst */
-        dev->buf[2] |= 0xcc;
+        dev->buf[2] |= IEEE802154_FCF_DST_ADDR_LONG;
         /* set destination address located directly after gnrc_ifhrd_t in memory */
-        memcpy(&(dev->buf)[index], gnrc_netif_hdr_get_dst_addr(hdr), 8);
-        index += 8;
-        /* set source pan_id, wireshark expects it there */
-        //dev->buf[index++] = (uint8_t)((dev->radio_pan) >> 8);
-        //dev->buf[index++] = (uint8_t)((dev->radio_pan) & 0xff);
-
-        /* set source address */
-        memcpy(&(dev->buf[index]), dev->addr_long, 8);
-        index += 8;
+        uint8_t *dst_addr = gnrc_netif_hdr_get_dst_addr(hdr);
+        for (int i = 7;  i >= 0; i--) {
+            dev->buf[index++] = dst_addr[i];
+        }
     }
     else {
         gnrc_pktbuf_release(pkt);
         return -ENOMSG;
     }
 
+    /* fill in source PAN ID (if applicable */
+    /* FIXME: compare with at86rf2xx, currently not working
```

Comment:
As far as I see, this option is never applied in the at86rf2xx driver by now. So compare with caution.